azurerm/internal/services/maintenance/parse/maintenance_configuration.go

```diff
@@ -0,0 +1,39 @@
+package parse
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type MaintenanceConfigurationId struct {
+	ResourceGroup string
+	Name          string
+}
+
+func MaintenanceConfigurationID(input string) (*MaintenanceConfigurationId, error) {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return nil, fmt.Errorf("[ERROR] Unable to parse MaintenanceConfiguration ID %q: %+v", input, err)
+	}
+
+	maintenanceConfiguration := MaintenanceConfigurationId{
+		ResourceGroup: id.ResourceGroup,
+	}
+
+	if name, err := id.PopSegment("maintenanceconfigurations"); err != nil {
+		if name, err = id.PopSegment("maintenanceConfigurations"); err != nil {
```

Comment:
We should open an SKD issue here and link to it

Reply:
thank you. I have added such bug about returned resource id:
https://github.com/Azure/azure-rest-api-specs/issues/8653
I have noted in the `resource_group_name` schema defnition